src/lib/util/Config.ts

```diff
@@ -123,21 +123,29 @@ export class Config {
 			return true;
 		}
 
-		return !(await this.getConfigValue('disabled', engine, this.typeChecker.booleanCheck));
+		return !(await this.getBooleanConfigValue('disabled', engine));
 	}
 
-	public async getSupportedLanguages(engine: ENGINE): Promise<string[]> {
-		return await this.getConfigValue('supportedLanguages', engine, this.typeChecker.stringArrayCheck);
+	public getSupportedLanguages(engine: ENGINE): Promise<any> {
+		return this.getStringArrayConfigValue('supportedLanguages', engine);
 	}
 
-	public async getTargetPatterns(engine: ENGINE): Promise<string[]> {
-		return await this.getConfigValue('targetPatterns', engine, this.typeChecker.stringArrayCheck);
+	public getTargetPatterns(engine: ENGINE): Promise<any> {
+		return this.getStringArrayConfigValue('targetPatterns', engine);
 	}
 
-	private async getConfigValue(propertyName: string, engine: ENGINE, typeChecker: (any, string, ENGINE) => boolean): Promise<string[]> {
+	protected getBooleanConfigValue(propertyName: string, engine: ENGINE): Promise<boolean> {
```

Comment:
It feels like you could remove the need for these two methods by using Generics (https://www.typescriptlang.org/docs/handbook/generics.html). E.g., declare your generic method as `protected getConfigValue<T>(propertyName: string, engine: ENGINE): Promise<T>`, and then invoke it as `getConfigValue<string>` or `getConfigValue<boolean>`.

Reply:
The main reason for the different implementations to avoid the caller needing to pass in a typechecker. #getBooelanConfigValue and #getStringArrayConfigValue need to pass different typecheckers to #getConfigValue. I don't think this can be achieved with a generic.

Reply:
I think if we use a mixture of generics and user-defined type guards, we could remove the need for these methods. https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards. I'm not certain though. Regardless, it's the sort of thing that could wait for a dedicated refactoring sprint.